Apps/System/settings.c

```diff
@@ -0,0 +1,196 @@
+}
+
+static void _bt_pair_window_load(Window *window) {
+    _bt_pair_accept = gbitmap_create_with_resource(RESOURCE_ID_CHECK_BLACK);
+    _bt_pair_reject = gbitmap_create_with_resource(RESOURCE_ID_DISMISS_BLACK);
+    GRect bmsize = gbitmap_get_bounds(_bt_pair_accept);
+    APP_LOG("settings", APP_LOG_LEVEL_INFO, "bt pair window load");
+    
+    window_set_background_color(window, GColorWhite);
+    
+    Layer *window_layer = window_get_root_layer(window);
+    GRect bounds = layer_get_unobstructed_bounds(window_layer);
+    
+    _bt_pair_layer = layer_create(bounds);
+    layer_add_child(window_layer, _bt_pair_layer);
+    layer_set_update_proc(_bt_pair_layer, _bt_pair_update_proc);
+    layer_mark_dirty(_bt_pair_layer);
+    
+    _bt_pair_action_bar = action_bar_layer_create();
+    action_bar_layer_set_click_config_provider(_bt_pair_action_bar, _bt_pair_action_click_config_provider);
+    action_bar_layer_set_icon_animated(_bt_pair_action_bar, BUTTON_ID_UP, _bt_pair_accept, true);
+    action_bar_layer_set_icon_animated(_bt_pair_action_bar, BUTTON_ID_DOWN, _bt_pair_reject, true);
+    action_bar_layer_set_background_color(_bt_pair_action_bar, GColorBlack);
+    action_bar_layer_add_to_window(_bt_pair_action_bar, window);
+}
+
+static void _bt_pair_window_unload(Window *window) {
+    action_bar_layer_destroy(_bt_pair_action_bar);
+    gbitmap_destroy(_bt_pair_accept);
+    gbitmap_destroy(_bt_pair_reject);
+    layer_destroy(_bt_pair_layer);
+    status_bar_layer_destroy(_bt_pair_status);
+    free(_bt_pair_name);
+}
+
+static Window *_main_window;
+static Menu *_menu;
+static StatusBarLayer *status_bar;
+
+static void _reset_menu_items(void);
+
+
+static void exit_to_watchface(struct Menu *menu, void *context)
+{
+    /* Exit to watchface */
+    hw_bluetooth_advertising_visible(0);
+    window_stack_pop(true);
+}
+
+static struct MenuItems *_really_wipe_fs(const struct MenuItem *ctx) {
+    fs_format();
+    panic("ok, goodbye");
+    return NULL;
+}
+
+static struct MenuItems *_wipe_fs(const struct MenuItem *ctx) {
+    MenuItems *items = menu_items_create(2);
+    menu_items_add(items, MenuItem("No", "j/k, lol", RESOURCE_ID_SPANNER, NULL));
+    menu_items_add(items, MenuItem("Yes", "I really mean it", RESOURCE_ID_SPANNER, _really_wipe_fs));
+    return items;
+}
+
+static struct MenuItems *_dummy_bt(const struct MenuItem *ctx) {
+    _bluetooth_pair_request(0, "Dummy phone name", NULL);
+    return NULL;
+}
+
+static struct MenuItems *_do_panic(const struct MenuItem *ctx) {
+    panic("Lolol!");
+    return NULL;
+}
+
+static void _reset_menu_items(void)
+{
+    MenuItems *back = _menu->items->back;
+    MenuIndex *index = &_menu->items->back_index;
+
+    MenuItems *items = menu_items_create(4);
+    
+    menu_items_add(items, MenuItem("Discoverable", hw_bluetooth_name(), RESOURCE_ID_SPANNER, NULL));
+    menu_items_add(items, MenuItem("Format filesystem", "Time to die, sucker!", RESOURCE_ID_SPANNER, _wipe_fs));
```

Comment:
#ifdef for platform support here actually

Reply:
See above.  If this isn't safe on snowy, let's fix it.